Tools/cases_generator/parser.py

```diff
@@ -74,16 +74,22 @@ class CacheEffect(Node):
 
 @dataclass
 class InstHeader(Node):
+    kind: Literal["inst", "op"]
```

Comment:
The "kind" tag used here and elsewhere seems a bit odd to me. Is it just more ergonomic than using subclassing?

Reply:
It's historic. I'm slowly getting rid of these.